cse/indexing/FileIndexer.py

```diff
@@ -39,49 +44,102 @@ def index(self):
                 try:
                     documentMap.get(data["commentId"])
                 except KeyError:
-                    tokens = self.__processComment(data["commentId"], data["comment_text"])
+                    # update document map
                     documentMap.insert(data["commentId"], lastPointer)
-                    self.__index.incDocumentCounter()
-                
+                    # index comment text tokens
+                    tokens = self.__processComment(pIndex, data["commentId"], data["comment_text"])
+                    pIndex.incDocumentCounter()
+                    # index comment parent-child structure
+                    cid = data["commentId"]
+                    parentCid = data["parent_comment_id"]
+                    if parentCid:
+                        rIndex.insert(parentCid, cid)
```

Comment:
This could lead to answering the request with it's own comment (one of the api's uses the parent field like the unix file system - root is it's own parent)

Reply:
Ok, I will change this.